### ISSUE TITLE
Improve exception message for missing dependent in table splitting

### ISSUE THREAD

**Issue by sbsw** [dotnet/efcore#14456]
Weird problem referencing owned types from classes that inherit from other classes.  I created a simple repro.  Please see https://stackoverflow.com/q/54246707/1637105

**Comment by ajcvickers** [comment]
Note for triage: simple repro:
```C#
public class IdentityUser
{
    public string Id { get; set; }
}

public class ApplicationUser : IdentityUser
{
    public PersonalName Name { get; set; }
    public ValidationToken ValidationToken { get; set; }
}

[Owned]
public class PersonalName
{
    public string FirstName { get; set; }
    public string LastName { get; set; }
}

[Owned]
public class ValidationToken
{
    public int ValidationCode { get; set; }
    public DateTime ExperiationDateUTC { get; set; }
}

public class BloggingContext : DbContext
{
    protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
    {
        optionsBuilder
            .UseSqlServer(@"Server=(localdb)\mssqllocaldb;Database=Test;ConnectRetryCount=0");
    }

    public DbSet<IdentityUser> Users { get; set; }
    public DbSet<ApplicationUser> AppUsers { get; set; }
}

public class Program
{
    public static void Main()
    {
        using (var context = new BloggingContext())
        {
            context.Database.EnsureDeleted();
            context.Database.EnsureCreated();
        }
    }
}
```

**Comment by sbsw** [comment]
Note the workaround in the [answer to the original SO post](https://stackoverflow.com/a/54335710/1637105).  The problem only happens when the relationship is configured via attribute markup.

**Comment by sbsw** [comment]
I thought that the workaround was to configure the owned relations in `OnModelCreating`, but I've added (and configured) a bunch of "owned" types to one of my inherited classes and now I get a new variant on the error:

> [InvalidOperationException] The entity of type 'Patient' is sharing the table 'AspNetUsers' with entities of type 'EmergencyContact', but there is no entity of this type with the same key value that has been marked as 'Added'. Consider using 'DbContextOptionsBuilder.EnableSensitiveDataLogging' to see the key values.

Note that Patient derives from AspNetUsers, and EmergencyContact is a POCO class that doesn't contain an Id property.

**Comment by sbsw** [comment]
Never mind... This is the error you get when you neglect to initialize the property that is a reference to the owned entity prior to adding the entity to the DB.

The error message is flat-out wrong. The error message should tell me that `Patient.EmergencyContact` can not be null. Or, if that level of detail isn't available, it should at least give me a clue that I need to initialize "owned" entity properties.

**Comment by AndriySvyryd** [comment]
Fixed in 6351aa1b42095aa939d5feb39a2041f299178f31
